frontend/src/components/MissionOverview/OngoingMissionCard.tsx

```diff
@@ -31,57 +32,76 @@ export function OngoingMissionCard({ mission }: MissionProps) {
 }
 
-function MissionControlButtons({mission}: MissionProps) {
-    const [status, setStatus] = useState(MissionStatus.Ongoing);
-    enum ControlButton {Pause, Stop, Play}
-    
+function MissionControlButtons({ mission }: MissionProps) {
+    const [status, setStatus] = useState(MissionStatus.Ongoing)
+    const apiCaller = useApi()
+    enum ControlButton {
+        Pause,
+        Stop,
+        Resume,
+    }
+
     const handleClick = (button: ControlButton) => {
-        switch(button){
-            case ControlButton.Pause:{
+        switch (button) {
+            case ControlButton.Pause: {
+                apiCaller.pauseMission(mission.robot.id)
                 setStatus(MissionStatus.Paused)
-                break;
+                break
             }
-            case ControlButton.Play:{
+            case ControlButton.Resume: {
+                apiCaller.resumeMission(mission.robot.id)
                 setStatus(MissionStatus.Ongoing)
-                break;
-
+                break
             }
-            case ControlButton.Stop:{
+            case ControlButton.Stop: {
+                apiCaller.resumeMission(mission.robot.id)
                 setStatus(MissionStatus.Cancelled)
-                break;
+                break
             }
         }
     }
 
     const renderControlIcon = (status: MissionStatus) => {
-        if(status == MissionStatus.Paused) {
-                return(
-                    <>
-                        <Icon name="play_circle" style={{ color: tokens.colors.interactive.primary__resting.rgba}} size={32} onClick={() => handleClick(ControlButton.Play)}/>
-                        <Icon name="stop" style={{ color: tokens.colors.interactive.primary__resting.rgba}} size={32} onClick={() => handleClick(ControlButton.Stop)}/>
-                    </>
-                )
+        if (status == MissionStatus.Paused) {
+            return (
+                <>
+                    <Icon
+                        name="play_circle"
+                        style={{ color: tokens.colors.interactive.primary__resting.rgba }}
+                        size={32}
+                        onClick={() => handleClick(ControlButton.Resume)}
+                    />
+                    <Icon
+                        name="stop"
+                        style={{ color: tokens.colors.interactive.primary__resting.rgba }}
+                        size={32}
+                        onClick={() => handleClick(ControlButton.Stop)}
+                    />
+                </>
+            )
         }
         return (
-            <Icon name="pause_circle" style={{ color: tokens.colors.interactive.primary__resting.rgba}} size={32} onClick={() => handleClick(ControlButton.Pause)}/>
+            <Icon
+                name="pause_circle"
+                style={{ color: tokens.colors.interactive.primary__resting.rgba }}
+                size={32}
+                onClick={() => handleClick(ControlButton.Pause)}
+            />
```

Comment:
Maybe we should group these Icons into named components "PauseButton", "PlayButton" and "StopButton"?